packages/priority-queue/src/main/PriorityQueue.ts

```diff
@@ -79,52 +93,43 @@ export class PriorityQueue {
     return this.queue.length;
   }
 
-  private resolveItems(): void {
+  private async processList(): Promise<void> {
     const queueObject = this.first;
     if (!queueObject) {
+      this.isRunning = false;
       return;
     }
 
-    /* tslint:disable:no-floating-promises */
-    Promise.resolve(queueObject.fn())
-      .then((result: any) => {
-        return {shouldContinue: true, wrappedResolve: () => queueObject.resolve(result)};
-      })
-      .catch((error: Error) => {
-        if (queueObject.retry > 0) {
-          queueObject.retry -= 1;
-          // TODO: Implement configurable reconnection delay (and reconnection delay growth factor)
-          setTimeout(() => this.resolveItems(), this.config.retryDelay || 1000);
-          return {shouldContinue: false};
-        } else {
-          queueObject.reject(error);
-          return {shouldContinue: true};
-        }
-      })
-      .then(({shouldContinue, wrappedResolve}: {shouldContinue: boolean; wrappedResolve?: Function}) => {
-        if (shouldContinue) {
-          if (wrappedResolve) {
-            wrappedResolve();
-          }
-          this.isPending = false;
-          this.queue.shift();
-          this.resolveItems();
-        }
-      });
-    /* tslint:enable:no-floating-promises */
+    try {
+      queueObject.resolve(await queueObject.fn());
+      this.queue.shift();
+      /* tslint:disable-next-line:no-floating-promises */
```

Comment:
What happens if the Promise rejects? Is that an issue we want to be warned about?